src/blueprints/auth/auth_routes.py

```diff
@@ -0,0 +1,71 @@
+from flask import render_template, request, redirect, url_for, session
+import logging
+from . import auth_blueprint
+from services import user_service
+from utils.form_handlers import UserHandlers
+from utils.csrf_utils import get_csrf_token
+
+
+@auth_blueprint.route('/')
+def index_route():
+    return render_template('index.html')
+
+
+@auth_blueprint.route('/login', methods=['GET'])
+def login_route_get():
+    error = session.pop('error', None)
+
+    csrf_token = get_csrf_token()
+    return render_template('login.html', error=error, csrf_token=csrf_token)
+    
+
+@auth_blueprint.route('/login', methods=['POST'])
+def login_route_post():
+    username, password = UserHandlers.get_login_form_data()
+
+    error = UserHandlers.login_form_check_input(username, password)
+
+    if not error:
+        try:
+            if user_service.login_user(username, password):
+                return redirect(url_for('auth_blueprint.index_route'))
+            else:
+                session['error'] = "Invalid credentials"
+                return redirect(url_for('auth_blueprint.login_route_get'))
+
+        except Exception as e:
+            session['error'] = "Login failed."
+            logging.error(f"Error occurred: {str(e)}")
+            return redirect(url_for('login_route_get'))
+    
+
+@auth_blueprint.route('/register', methods=['GET'])
+def register_route_get():
+    error = None
+
+    csrf_token = get_csrf_token()
+    return render_template('register.html', error=error, csrf_token=csrf_token)
+
+
+@auth_blueprint.route('/register', methods=['POST'])
+def register_route_post():
+    error = None
+
+    username, role, password_a, password_b = UserHandlers.get_register_form_data()
+
+    error = UserHandlers.register_form_check_input(username, password_a, password_b, error)
```

Comment:
What is the difference between, for example, UserHandlers and character_service? Should they be named similarly, or do they have different type of responsibility?

Reply:
They could be named more clearly, yes. The handlers take care of the forms, namely login, register and character creation forms. It could also be implied better in their names. They include functions that fetch and validate form data. They are grouped in Utils. The services contain the actual database operations. Somebody already commented on the project structure earlier, which led me to switch the name from components to services.